label/reports/dynamic_label.xml

```diff
@@ -0,0 +1,80 @@
+<openerp>
+  <data>
+    <template id="report_label">
+      <style type="text/css">
+	div.o_label_div {
+	overflow: hidden;
+	width:100%;
+	height:100%;
+	}
+	table.o_label_table {
+	width: 100%; height: 100%;
+	}
+      </style>
+      <t t-call="report.html_container">
+	<t t-foreach="docs" t-as="o">
+	  <t t-call="report.external_layout">
+	    <div class="page">
+	      <t t-foreach="range(len(get_data(rows, columns, ids, model, o.number_of_labels)))" t-as="data_page">
+		<div  t-attf-style="position: absolute; left: {{ o.name.left_margin * o.mm2px}}px; top: {{(297 * data_page + o.name.top_margin)*o.mm2px}}px; height: {{(297 - o.name.top_margin - o.name.bottom_margin)*o.mm2px}}px; width: {{o.mm2px*(210 - o.name.left_margin - o.name.right_margin)}}px;">
+		  <!-- Start page -->
+		  <table class="o_label_table" t-attf-style="border-space: 1px {{o.name.cell_spacing*o.mm2px}}px;">
+		    <t t-foreach="get_data(rows, columns, ids, model, o.number_of_labels)[data_page]" t-as="row">
+		      <tr>
+			<t t-foreach="row" t-as="col">
+			  <td t-attf-style="height: {{o.name.height * o.mm2px}}px; width: {{ o.name.width * o.mm2px }}">
+			    
+			    <!-- One Label -->
+			    <!-- In order to keep the layout whatever the datas are, everything is inside a div that hide the overflow and occupy all the place -->
+			    <div class="o_label_div" t-attf-style="padding-right: {{padding_right*o.mm2px}}px; padding-bottom: {{padding_bottom*o.mm2px}}px; padding-top: {{padding_top*o.mm2px}}px; padding-left: {{padding_left*o.mm2px}}px;">
```

Comment:
Please check if this can be refactored into a separate template, this would avoid the overly high level of indentation below